### DISCUSSION TITLE
When connecting via IKEv2 (EAP-md5 certificate + user/pass) from android strongswan, received error message from my router which is hosting  (Auth payload must not be included)

### DISCUSSION THREAD

**Discussion by arthurOmanko** [strongswan/strongswan#2451] [Q&A]
When connecting via IKEv2 (EAP-md5 (EAP) certificate + user/pass) from android strongswan app, received error message from my router which is hosting  (Auth payload must not be included)
my Router requesting "auth payload not included" on document
How can I omit Auth payload in IKE_Auth exchange message when using android strongswan app (EAP certificate + user/pass)?

**Comment by tobiasbrunner** [comment]
You have to use `IKEv2 EAP (Username/Password)`, otherwise, the client will try to authenticate using a certificate first, followed by an EAP authentication as a second authentication round (other than strongSwan there are not many IKEv2 implementations that support multiple authentication rounds).

**Reply by arthurOmanko** [reply]
> You have to use `IKEv2 EAP (Username/Password)`, otherwise, the client will try to authenticate using a certificate first, followed by an EAP authentication as a second authentication round (other than strongSwan there are not many IKEv2 implementations that support multiple authentication rounds).

I tried IKEv2 EAP (Username/Password). and I got error (no EAP key found for hosts `www.aaa.com` - '%any'.   EAP_MD5 method failed, which are messages on strongswap app)
Of course I input username/password.
why your soft cant find EAP key? 
Server requested EAP_MD5 authentification (id 0x0F) so I think on the way some successed.
I created not-true www.aaa.com as client identity. it reason?
I must use mail address as client identity?
Or your soft bug?
currently I could connect to server when only certificates. (server identity = server fqdn, client identity = www.aaa.com (not-true)
but I could not connect to it when IKEv2 EAP (Username/Password)

**Reply by tobiasbrunner** [reply]
Does the server request the username with an EAP-Identity exchange before starting EAP-MD5? If not, then you have to configure the username as IKE identity (or not configure it as that will be the default).
